catroid/src/main/java/org/catrobat/catroid/content/actions/PauseForBeatsAction.kt

```diff
@@ -20,44 +20,37 @@
  * You should have received a copy of the GNU Affero General Public License
  * along with this program.  If not, see <http://www.gnu.org/licenses/>.
  */
-package org.catrobat.catroid.content.actions;
-
-import android.util.Log;
-
-import com.badlogic.gdx.scenes.scene2d.actions.TemporalAction;
-
-import org.catrobat.catroid.content.Scope;
-import org.catrobat.catroid.formulaeditor.Formula;
-import org.catrobat.catroid.formulaeditor.InterpretationException;
-import org.catrobat.catroid.pocketmusic.mididriver.MidiSoundManager;
-
-public class PauseForBeatsAction extends TemporalAction {
-
-	private Scope scope;
-	private Formula beats;
-
-	@Override
-	protected void begin() {
-		try {
-			float pausedBeats = 0;
-			if (beats != null) {
-				pausedBeats = beats.interpretFloat(scope);
-			}
-			super.setDuration((float) MidiSoundManager.getInstance().getDurationForBeats(pausedBeats) / 1000);
-		} catch (InterpretationException interpretationException) {
-			Log.d(getClass().getSimpleName(), "Formula interpretation for this specific Brick failed.", interpretationException);
-		}
-	}
-
-	public void setScope(Scope scope) {
-		this.scope = scope;
-	}
-
-	public void setBeats(Formula beats) {
-		this.beats = beats;
-	}
-
-	@Override
-	protected void update(float percent) {
-	}
+package org.catrobat.catroid.content.actions
+
+import android.util.Log
+import com.badlogic.gdx.scenes.scene2d.actions.TemporalAction
+import org.catrobat.catroid.common.Constants
+import org.catrobat.catroid.content.Scope
+import org.catrobat.catroid.formulaeditor.Formula
+import org.catrobat.catroid.formulaeditor.InterpretationException
+import org.catrobat.catroid.pocketmusic.mididriver.MidiSoundManager
+
+class PauseForBeatsAction : TemporalAction() {
+    private lateinit var scope: Scope
+    private lateinit var beats: Formula
+
+    override fun begin() {
+        try {
+            val pausedBeats: Float = beats.interpretFloat(scope)
+            super.setDuration(MidiSoundManager.getInstance().getDurationForBeats(pausedBeats).toFloat() / Constants.MILLISECONDS_IN_A_SECOND)
+        } catch (interpretationException: InterpretationException) {
+            Log.d(javaClass.simpleName, "Formula interpretation for this specific Brick failed.", interpretationException)
+        }
+    }
+
+    fun setScope(scope: Scope) {
+        this.scope = scope
+    }
+
+    fun setBeats(beats: Formula) {
+        this.beats = beats
+    }
+
+    @SuppressWarnings("EmptyFunctionBlock")
```

Comment:
you get the warning here because you can write:
override fun update(percent: Float) = Unit
Please change it to that.